sphinx_gallery/gen_rst.py

```diff
@@ -654,3 +657,45 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
         logger.debug("%s ran in : %.2g seconds", src_file, time_elapsed)
 
     return intro, time_elapsed
+
+
+def gen_binder_url(fname, binder_conf):
+    if not isinstance(binder_conf, dict) or len(binder_conf) == 0:
+        return None
+    req_values = ['url', 'org', 'repo', 'branch', 'dependencies']
+
+    # Ensure all fields are populated
+    missing_values = [val for val in req_values
+                      if not isinstance(binder_conf.get(val, None),
+                                        (str, unicode))]
+    if len(missing_values) > 0:
+        raise ValueError('binder_conf is missing values for: {}'.format(
+            missing_values))
```

Comment:
L663-673, I would like to isolate this in a separate function. Then write a unittest.